src/function/gds/K_Core_Decomposition.cpp

```diff
@@ -0,0 +1,192 @@
+#include "binder/binder.h"
+#include "common/types/types.h"
+#include "function/gds/gds_frontier.h"
+#include "function/gds/gds_function_collection.h"
+#include "function/gds/gds_utils.h"
+#include "function/gds/rec_joins.h"
+#include "function/gds_function.h"
+#include "graph/graph.h"
+#include "processor/execution_context.h"
+#include "processor/result/factorized_table.h"
+
+using namespace kuzu::binder;
+using namespace kuzu::common;
+using namespace kuzu::processor;
+using namespace kuzu::storage;
+using namespace kuzu::graph;
+
+namespace kuzu {
+namespace function {
+struct KCoreInitEdgeCompute : public EdgeCompute {
+    KCoreFrontierPair* frontierPair;
+
+    explicit KCoreInitEdgeCompute(KCoreFrontierPair* frontierPair) : frontierPair{frontierPair} {}
+
+    std::vector<common::nodeID_t> edgeCompute(common::nodeID_t boundNodeID, graph::NbrScanState::Chunk& chunk, bool) override {
+        std::vector<common::nodeID_t> result;
+        uint64_t nbrAmount = 0;
+        chunk.forEach([&](auto nbrNodeID, auto) {nbrAmount++;
+        result.push_back(nbrNodeID);});
```

Comment:
why do we need to put them in result? As I understand, this EdgeCompute is to init degree and should run for only 1 iteration.